src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/bfaction/BfActionReport.java

```diff
@@ -0,0 +1,116 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.reports.bfaction;
+
+import com.google.common.base.Strings;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.ecm.common.helpers.UtilHelper;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ecm.common.model.ccd.items.BFActionTypeItem;
+import uk.gov.hmcts.ecm.common.model.listing.ListingData;
+import uk.gov.hmcts.ecm.common.model.listing.ListingDetails;
+import uk.gov.hmcts.ecm.common.model.listing.items.BFDateTypeItem;
+import uk.gov.hmcts.ecm.common.model.listing.items.BFDateTypeItemComparator;
+import uk.gov.hmcts.ecm.common.model.listing.types.BFDateType;
+import uk.gov.hmcts.ethos.replacement.docmosis.helpers.ReportHelper;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.BROUGHT_FORWARD_REPORT;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.OLD_DATE_TIME_PATTERN;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.OLD_DATE_TIME_PATTERN2;
+
+@Service
+public class BfActionReport {
+    private static final String SPACE = " ";
+    private static final String DATE_TIME_SEPARATOR = "T";
+    private static final String MILLISECOND_PART = ".000";
+
+    public ListingData runReport(ListingDetails listingDetails, List<SubmitEvent> submitEvents) {
+        BfActionReportData bfActionReportData = new BfActionReportData();
+        if (!CollectionUtils.isEmpty(submitEvents)) {
+            List<BFDateTypeItem> bfDateTypeItems = new ArrayList<>();
+            for (SubmitEvent submitEvent : submitEvents) {
+                addBfDateTypeItems(submitEvent, listingDetails.getCaseData(), bfDateTypeItems);
+            }
+            bfDateTypeItems.sort(new BFDateTypeItemComparator());
+            bfActionReportData.setBfDateCollection(bfDateTypeItems);
+        }
+
+        String caseTypeId = listingDetails.getCaseTypeId();
+        var office = UtilHelper.getListingCaseTypeId(caseTypeId);
+        bfActionReportData.clearReportFields();
+        bfActionReportData.setReportType(listingDetails.getCaseData().getReportType());
+        bfActionReportData.setReportType(BROUGHT_FORWARD_REPORT);
+        bfActionReportData.setDocumentName(BROUGHT_FORWARD_REPORT);
+        bfActionReportData.setOffice(office);
+        bfActionReportData.setDurationDescription(ReportHelper.getDurationText(listingDetails.getCaseData()));
+        return bfActionReportData;
```

Comment:
The reason why I did it this way is actually because the templates that expect date range do not look right when the search is for a single date. For example, please have a look at the Claims Accepted report(template number 002219).

Reply:
This was due to an issue in the template but has now been resolved. For the date section, you need to make sure it is 
`Period: <<cs_{isBlank(Listed_date)}>>Between <<Listed_date_from>> and <<Listed_date_to>><<else>><<Listed_date>><<es_>>`. Claims Accepted and Live Caseload were ones James noticed when he was reviewing reports

Reply:
That is another way of solving the issue.